src/System.IO.Ports/tests/Support/PortHelper.cs

```diff
@@ -22,9 +23,36 @@ public static string[] GetPorts()
         {
             if (PlatformDetection.IsUap)
             {
-                return new [] { "COM3", "COM4", "COM5", "COM6", "COM7" }; // we are waiting for a Win32 new QueryDosDevice API since the current doesn't work for Uap https://github.com/dotnet/corefx/issues/21156
+                // On UAP it is not possible to call QueryDosDevice, so use HARDWARE\DEVICEMAP\SERIALCOMM on the registry
+                // to get this information. The UAP code uses the GetCommPorts API to retrieve the same information.
+                return GetCommPortsFromRegistry();
             }
 
+            return GetCommPortsViaQueryDosDevice();
+        }
+
+        private static string[] GetCommPortsFromRegistry()
+        {
+            using (RegistryKey serialKey = Registry.LocalMachine.OpenSubKey(@"HARDWARE\DEVICEMAP\SERIALCOMM"))
```

Comment:
Might be worth adding the link to https://msdn.microsoft.com/en-us/library/windows/hardware/ff546502.aspx.